### PR TITLE
qcut() converted into FExpr

### PR DESCRIPTION
**Pull request by st-pasha**
WIP for #2562

### CODE REVIEW

File: src/core/expr/fexpr_qcut.cc [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+equal populations. Thus, the intervals are chosen according to
+the sample quantiles of the data.
+
+If there are duplicate values in the data, they will all be placed
+into the same been. In extreme cases this may cause the bins to be
+highly unbalanced.
+
+Parameters
+----------
+cols: FExpr
```
Comment (oleksiyskononenko) [inline src/core/expr/fexpr_qcut.cc:153]: I remember we discussed that the input can be also different from `FExpr`?

File: src/core/expr/fexpr_qcut.cc [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    Input data for quantile binning.
+
+nquantiles: int | List[int]
```
Comment (oleksiyskononenko) [inline src/core/expr/fexpr_qcut.cc:156]: Can also be a tuple...

Comment (st-pasha) [reply src/core/expr/fexpr_qcut.cc:156]: True, anywhere where we accept a list, we also accept a tuple. 
However, when I tried to explicitly state this in argument's type, it quickly became a mess. Not here (this is a simple case), but in some other places where we have for example `List[bool | int | float | str]`. So I decided that it's just easier to document that something is a list, and leave tuple as "an extra thing that also happens to work".

File: src/core/expr/fexpr_qcut.cc (hunk 1 above)
Comment (st-pasha) [reply src/core/expr/fexpr_qcut.cc:153]: Yeah, we'll change this (and other places) once we have a documentation for what it means to be "FExpr-like".

**Pull request merged**